### ISSUE TITLE
💥 Crash on ` -[UIView(PNDViewTree) makeNode:parentId:ancestorVC:depth:options:]`

### ISSUE THREAD

**Issue by msrutek-paylocity** [pendo-io/pendo-mobile-sdk#152]
## **Platform + Version**
iOS 17.5.1
iPhone 11

## **SDK Version**
3.2.2

## **Framework**
Native, SwiftUI

## **Describe the bug**
Crash of the app.

## **To Reproduce**
Yet unknown.

## **Expected behavior**
No crash.

## Stack Trace

```
Crashed: com.apple.main-thread
0  libobjc.A.dylib                0xa418 append_referrer(weak_entry_t*, objc_object**) + 76
1  libobjc.A.dylib                0x9ffc weak_register_no_lock + 184
2  libobjc.A.dylib                0x9ce0 objc_storeWeak + 472
3  Pendo                          0x8fe74 -[UIView(PNDViewTree) makeNode:parentId:ancestorVC:depth:options:] + 120
4  Pendo                          0xab7a0 -[PNDViewTree makeNode:parentId:assignedId:depth:parentVCAncestor:nodeOptions:] + 176
5  Pendo                          0xab478 -[PNDViewTree recursiveSubTreeDFS:parentId:depth:parentVCAncestor:] + 152
6  Pendo                          0xab58c -[PNDViewTree recursiveSubTreeDFS:parentId:depth:parentVCAncestor:] + 428
7  Pendo                          0xab58c -[PNDViewTree recursiveSubTreeDFS:parentId:depth:parentVCAncestor:] + 428
8  Pendo                          0xab58c -[PNDViewTree recursiveSubTreeDFS:parentId:depth:parentVCAncestor:] + 428
9  Pendo                          0xab58c -[PNDViewTree recursiveSubTreeDFS:parentId:depth:parentVCAncestor:] + 428
10 Pendo                          0xab58c -[PNDViewTree recursiveSubTreeDFS:parentId:depth:parentVCAncestor:] + 428
11 Pendo                          0xab320 -[PNDViewTree viewTree] + 424
12 Pendo                          0xaa0b8 +[PNDViewTree treeWithRoot:options:] + 268
13 Pendo                          0x11db30 -[PNDScreenManager retroTreeArrayFor:] + 24
14 Pendo                          0x11daac -[PNDScreenManager textsScreenData:dict:] + 112
15 Pendo                          0x11d7b8 -[PNDScreenManager screenDataFor:] + 252
16 Pendo                          0x11d4b4 -[PNDScreenManager screenChanged] + 56
17 Pendo                          0x11ce90 __44-[PNDScreenManager triggerScreenScanOfType:]_block_invoke.50 + 508
18 Pendo                          0x10f9b8 -[PNDDebouncer fireNow] + 36
19 Foundation                     0x1b4f1c __NSFireTimer + 96
20 CoreFoundation                 0xb7a2c __CFRUNLOOP_IS_CALLING_OUT_TO_A_TIMER_CALLBACK_FUNCTION__ + 32
21 CoreFoundation                 0xb76d0 __CFRunLoopDoTimer + 1004
22 CoreFoundation                 0xb722c __CFRunLoopDoTimers + 288
23 CoreFoundation                 0x53888 __CFRunLoopRun + 1856
24 CoreFoundation                 0x52cd8 CFRunLoopRunSpecific + 608
25 GraphicsServices               0x11a8 GSEventRunModal + 164
26 UIKitCore                      0x40a90c -[UIApplication _run] + 888
27 UIKitCore                      0x4be9d0 UIApplicationMain + 340
28 REDACTED                       0x584c main + 9 (AppDelegate.swift:9)
29 ???                            0x1b3b99e4c (Missing)
```

**Comment by MaximShnirman** [comment]
hi @msrutek-paylocity, can you please share the full crash log?

**Comment by msrutek-paylocity** [comment]
@MaximShnirman hi, the full crash log can be found below:

<details>
<summary>Crash log >>></summary>

```
# Crashlytics - Stack trace
# Platform: apple
# Version: 24.6.0 (5371)
# Date: Fri Jul 05 2024 14:49:01 GMT+0200 (Central European Summer Time)

Crashed: com.apple.main-thread
0  libobjc.A.dylib                0xa418 append_referrer(weak_entry_t*, objc_object**) + 76
1  libobjc.A.dylib                0x9ffc weak_register_no_lock + 184
2  libobjc.A.dylib                0x9ce0 objc_storeWeak + 472
3  Pendo                          0x8fe74 -[UIView(PNDViewTree) makeNode:parentId:ancestorVC:depth:options:] + 120
4  Pendo                          0xab7a0 -[PNDViewTree makeNode:parentId:assignedId:depth:parentVCAncestor:nodeOptions:] + 176
5  Pendo                          0xab478 -[PNDViewTree recursiveSubTreeDFS:parentId:depth:parentVCAncestor:] + 152
6  Pendo                          0xab58c -[PNDViewTree recursiveSubTreeDFS:parentId:depth:parentVCAncestor:] + 428
7  Pendo                          0xab58c -[PNDViewTree recursiveSubTreeDFS:parentId:depth:parentVCAncestor:] + 428
8  Pendo                          0xab58c -[PNDViewTree recursiveSubTreeDFS:parentId:depth:parentVCAncestor:] + 428
9  Pendo                          0xab58c -[PNDViewTree recursiveSubTreeDFS:parentId:depth:parentVCAncestor:] + 428
10 Pendo                          0xab58c -[PNDViewTree recursiveSubTreeDFS:parentId:depth:parentVCAncestor:] + 428
11 Pendo                          0xab320 -[PNDViewTree viewTree] + 424
12 Pendo                          0xaa0b8 +[PNDViewTree treeWithRoot:options:] + 268
13 Pendo                          0x11db30 -[PNDScreenManager retroTreeArrayFor:] + 24
14 Pendo                          0x11daac -[PNDScreenManager textsScreenData:dict:] + 112
15 Pendo                          0x11d7b8 -[PNDScreenManager screenDataFor:] + 252
16 Pendo                          0x11d4b4 -[PNDScreenManager screenChanged] + 56
17 Pendo                          0x11ce90 __44-[PNDScreenManager triggerScreenScanOfType:]_block_invoke.50 + 508
18 Pendo                          0x10f9b8 -[PNDDebouncer fireNow] + 36
19 Foundation                     0x1b4f1c __NSFireTimer + 96
20 CoreFoundation                 0xb7a2c __CFRUNLOOP_IS_CALLING_OUT_TO_A_TIMER_CALLBACK_FUNCTION__ + 32
21 CoreFoundation                 0xb76d0 __CFRunLoopDoTimer + 1004
22 CoreFoundation                 0xb722c __CFRunLoopDoTimers + 288
23 CoreFoundation                 0x53888 __CFRunLoopRun + 1856
24 CoreFoundation                 0x52cd8 CFRunLoopRunSpecific + 608
25 GraphicsServices               0x11a8 GSEventRunModal + 164
26 UIKitCore                      0x40a90c -[UIApplication _run] + 888
27 UIKitCore                      0x4be9d0 UIApplicationMain + 340
28 REDACTED                       0x584c main + 9 (AppDelegate.swift:9)
29 ???                            0x1b3b99e4c (Missing)

com.apple.runningboardservices.connection.incoming
0  libsystem_kernel.dylib         0x161c __ulock_wait2 + 8
1  libsystem_platform.dylib       0x34d8 _os_unfair_lock_lock_slow + 188
2  libobjc.A.dylib                0xa5cc locker_mixin<lockdebug::lock_mixin<objc_lock_base_t> >::lockWith(lockdebug::lock_mixin<objc_lock_base_t>&) + 116
3  libobjc.A.dylib                0x9b9c objc_storeWeak + 148
4  Foundation                     0xeffc clearARCWeakAt + 40
5  Foundation                     0xb9c64 -[NSConcreteMapTable removeObjectForKey:] + 220
6  RunningBoardServices           0x31a70 -[RBSConnection async_assertionsDidInvalidate:withError:] + 288
7  RunningBoardServices           0x325c8 __32-[RBSConnection _handleMessage:]_block_invoke.212 + 272
8  libsystem_trace.dylib          0x194c _os_activity_initiate_impl + 64
9  RunningBoardServices           0xef04 -[RBSConnection _handleMessage:] + 1000
10 RunningBoardServices           0xe95c __37-[RBSConnection _lock_setConnection:]_block_invoke + 80
11 libxpc.dylib                   0x11cbc _xpc_connection_call_event_handler + 144
12 libxpc.dylib                   0x13908 _xpc_connection_mach_event + 1404
13 libdispatch.dylib              0x3e94 _dispatch_client_callout4 + 20
14 libdispatch.dylib              0x20000 _dispatch_mach_msg_invoke + 468
15 libdispatch.dylib              0xb284 _dispatch_lane_serial_drain + 368
16 libdispatch.dylib              0x20d50 _dispatch_mach_invoke + 444
17 libdispatch.dylib              0xb284 _dispatch_lane_serial_drain + 368
18 libdispatch.dylib              0xbf64 _dispatch_lane_invoke + 432
19 libdispatch.dylib              0x16cb4 _dispatch_root_queue_drain_deferred_wlh + 288
20 libdispatch.dylib              0x16528 _dispatch_workloop_worker_thread + 404
21 libsystem_pthread.dylib        0x4934 _pthread_wqthread + 288
22 libsystem_pthread.dylib        0x10cc start_wqthread + 8

com.apple.uikit.eventfetch-thread
0  libsystem_kernel.dylib         0x1808 mach_msg2_trap + 8
1  libsystem_kernel.dylib         0x5008 mach_msg2_internal + 80
2  libsystem_kernel.dylib         0x4f20 mach_msg_overwrite + 436
3  libsystem_kernel.dylib         0x4d60 mach_msg + 24
4  CoreFoundation                 0x53f5c __CFRunLoopServiceMachPort + 160
5  CoreFoundation                 0x53600 __CFRunLoopRun + 1208
6  CoreFoundation                 0x52cd8 CFRunLoopRunSpecific + 608
7  Foundation                     0xc7e4c -[NSRunLoop(NSRunLoop) runMode:beforeDate:] + 212
8  Foundation                     0xc7c9c -[NSRunLoop(NSRunLoop) runUntilDate:] + 64
9  UIKitCore                      0x41e640 -[UIEventFetcher threadMain] + 420
10 Foundation                     0xde718 __NSThread__start__ + 732
11 libsystem_pthread.dylib        0x606c _pthread_start + 136
12 libsystem_pthread.dylib        0x10d8 thread_start + 8

com.apple.root.utility-qos
0  libswiftCore.dylib             0x3f9950 swift_conformsToProtocolMaybeInstantiateSuperclasses(swift::TargetMetadata<swift::InProcess> const*, swift::TargetProtocolDescriptor<swift::InProcess> const*, bool)::$_8::operator()((anonymous namespace)::ConformanceSection const&) const::'lambda'(swift::TargetProtocolConformanceDescriptor<swift::InProcess> const&)::operator()(swift::TargetProtocolConformanceDescriptor<swift::InProcess> const&) const + 740
1  libswiftCore.dylib             0x3f87bc swift_conformsToProtocolMaybeInstantiateSuperclasses(swift::TargetMetadata<swift::InProcess> const*, swift::TargetProtocolDescriptor<swift::InProcess> const*, bool) + 3948
2  libswiftCore.dylib             0x3f6b48 swift_conformsToProtocolCommon + 136
3  libswiftCore.dylib             0x39a9f0 swift::_conformsToProtocol(swift::OpaqueValue const*, swift::TargetMetadata<swift::InProcess> const*, swift::TargetProtocolDescriptorRef<swift::InProcess>, swift::TargetWitnessTable<swift::InProcess> const**) + 60
4  libswiftCore.dylib             0x3f5698 swift::_checkGenericRequirements(__swift::__runtime::llvm::ArrayRef<swift::TargetGenericRequirementDescriptor<swift::InProcess> >, __swift::__runtime::llvm::SmallVectorImpl<void const*>&, std::__1::function<void const* (unsigned int, unsigned int)>, std::__1::function<swift::TargetWitnessTable<swift::InProcess> const* (swift::TargetMetadata<swift::InProcess> const*, unsigned int)>) + 7484
5  libswiftCore.dylib             0x3de9b4 _gatherGenericParameters(swift::TargetContextDescriptor<swift::InProcess> const*, __swift::__runtime::llvm::ArrayRef<swift::MetadataOrPack>, swift::TargetMetadata<swift::InProcess> const*, __swift::__runtime::llvm::SmallVectorImpl<unsigned int>&, __swift::__runtime::llvm::SmallVectorImpl<void const*>&, swift::Demangle::__runtime::Demangler&) + 1860
6  libswiftCore.dylib             0x3e8f4c (anonymous namespace)::DecodedMetadataBuilder::createBoundGenericType(swift::TargetContextDescriptor<swift::InProcess> const*, __swift::__runtime::llvm::ArrayRef<swift::MetadataOrPack>, swift::MetadataOrPack) const + 248
7  libswiftCore.dylib             0x3e7270 swift::Demangle::__runtime::TypeDecoder<(anonymous namespace)::DecodedMetadataBuilder>::decodeMangledType(swift::Demangle::__runtime::Node*, unsigned int, bool) + 12276
8  libswiftCore.dylib             0x3e1110 swift_getTypeByMangledNodeImpl(swift::MetadataRequest, swift::Demangle::__runtime::Demangler&, swift::Demangle::__runtime::Node*, void const* const*, std::__1::function<void const* (unsigned int, unsigned int)>, std::__1::function<swift::TargetWitnessTable<swift::InProcess> const* (swift::TargetMetadata<swift::InProcess> const*, unsigned int)>) + 892
9  libswiftCore.dylib             0x3e0cb4 swift_getTypeByMangledNode + 836
10 libswiftCore.dylib             0x3e1774 swift_getTypeByMangledNameImpl(swift::MetadataRequest, __swift::__runtime::llvm::StringRef, void const* const*, std::__1::function<void const* (unsigned int, unsigned int)>, std::__1::function<swift::TargetWitnessTable<swift::InProcess> const* (swift::TargetMetadata<swift::InProcess> const*, unsigned int)>) + 1188
11 libswiftCore.dylib             0x3db610 swift_getTypeByMangledName + 836
12 libswiftCore.dylib             0x3dba58 swift_getTypeByMangledNameInContextImpl(char const*, unsigned long, swift::TargetContextDescriptor<swift::InProcess> const*, void const* const*) + 172
13 AttributeGraph                 0x546c AG::swift::metadata::mangled_type_name_ref(char const*, bool, AG::swift::metadata::ref_kind*) const + 212
14 AttributeGraph                 0x52f4 AG::swift::metadata_visitor::visit_field(AG::swift::metadata const*, AG::swift::field_record const&, unsigned long, unsigned long) + 80
15 AttributeGraph                 0x5220 AG::swift::metadata::visit(AG::swift::metadata_visitor&) const + 1244
16 AttributeGraph                 0x4c54 AG::LayoutDescriptor::make_layout(AG::swift::metadata const*, AGComparisonMode, AG::LayoutDescriptor::HeapMode) + 580
17 AttributeGraph                 0x4140 AG::(anonymous namespace)::TypeDescriptorCache::fetch(AG::swift::metadata const*, unsigned int, AG::LayoutDescriptor::HeapMode, int) + 344
18 AttributeGraph                 0x5928 AG::LayoutDescriptor::Builder::should_visit_fields(AG::swift::metadata const*, bool) + 56
19 AttributeGraph                 0x650c AG::LayoutDescriptor::Builder::visit_case(AG::swift::metadata const*, AG::swift::field_record const&, unsigned int) + 428
20 AttributeGraph                 0x511c AG::swift::metadata::visit(AG::swift::metadata_visitor&) const + 984
21 AttributeGraph                 0x4c54 AG::LayoutDescriptor::make_layout(AG::swift::metadata const*, AGComparisonMode, AG::LayoutDescriptor::HeapMode) + 580
22 AttributeGraph                 0x4140 AG::(anonymous namespace)::TypeDescriptorCache::fetch(AG::swift::metadata const*, unsigned int, AG::LayoutDescriptor::HeapMode, int) + 344
23 AttributeGraph                 0x5928 AG::LayoutDescriptor::Builder::should_visit_fields(AG::swift::metadata const*, bool) + 56
24 AttributeGraph                 0x57cc AG::LayoutDescriptor::Builder::visit_element(AG::swift::metadata const*, AG::swift::metadata::ref_kind, unsigned long, unsigned long) + 120
25 AttributeGraph                 0x5358 AG::swift::metadata_visitor::visit_field(AG::swift::metadata const*, AG::swift::field_record const&, unsigned long, unsigned long) + 180
26 AttributeGraph                 0x5220 AG::swift::metadata::visit(AG::swift::metadata_visitor&) const + 1244
27 AttributeGraph                 0x4c54 AG::LayoutDescriptor::make_layout(AG::swift::metadata const*, AGComparisonMode, AG::LayoutDescriptor::HeapMode) + 580
28 AttributeGraph                 0x4140 AG::(anonymous namespace)::TypeDescriptorCache::fetch(AG::swift::metadata const*, unsigned int, AG::LayoutDescriptor::HeapMode, int) + 344
29 AttributeGraph                 0x5928 AG::LayoutDescriptor::Builder::should_visit_fields(AG::swift::metadata const*, bool) + 56
30 AttributeGraph                 0x650c AG::LayoutDescriptor::Builder::visit_case(AG::swift::metadata const*, AG::swift::field_record const&, unsigned int) + 428
31 AttributeGraph                 0x511c AG::swift::metadata::visit(AG::swift::metadata_visitor&) const + 984
32 AttributeGraph                 0x4c54 AG::LayoutDescriptor::make_layout(AG::swift::metadata const*, AGComparisonMode, AG::LayoutDescriptor::HeapMode) + 580
33 AttributeGraph                 0x4140 AG::(anonymous namespace)::TypeDescriptorCache::fetch(AG::swift::metadata const*, unsigned int, AG::LayoutDescriptor::HeapMode, int) + 344
34 AttributeGraph                 0x5928 AG::LayoutDescriptor::Builder::should_visit_fields(AG::swift::metadata const*, bool) + 56
35 AttributeGraph                 0x57cc AG::LayoutDescriptor::Builder::visit_element(AG::swift::metadata const*, AG::swift::metadata::ref_kind, unsigned long, unsigned long) + 120
36 AttributeGraph                 0x5358 AG::swift::metadata_visitor::visit_field(AG::swift::metadata const*, AG::swift::field_record const&, unsigned long, unsigned long) + 180
37 AttributeGraph                 0x5220 AG::swift::metadata::visit(AG::swift::metadata_visitor&) const + 1244
38 AttributeGraph                 0x4c54 AG::LayoutDescriptor::make_layout(AG::swift::metadata const*, AGComparisonMode, AG::LayoutDescriptor::HeapMode) + 580
39 AttributeGraph                 0x4140 AG::(anonymous namespace)::TypeDescriptorCache::fetch(AG::swift::metadata const*, unsigned int, AG::LayoutDescriptor::HeapMode, int) + 344
40 AttributeGraph                 0x5928 AG::LayoutDescriptor::Builder::should_visit_fields(AG::swift::metadata const*, bool) + 56
41 AttributeGraph                 0x650c AG::LayoutDescriptor::Builder::visit_case(AG::swift::metadata const*, AG::swift::field_record const&, unsigned int) + 428
42 AttributeGraph                 0x511c AG::swift::metadata::visit(AG::swift::metadata_visitor&) const + 984
43 AttributeGraph                 0x4c54 AG::LayoutDescriptor::make_layout(AG::swift::metadata const*, AGComparisonMode, AG::LayoutDescriptor::HeapMode) + 580
44 AttributeGraph                 0x4140 AG::(anonymous namespace)::TypeDescriptorCache::fetch(AG::swift::metadata const*, unsigned int, AG::LayoutDescriptor::HeapMode, int) + 344
45 AttributeGraph                 0x5928 AG::LayoutDescriptor::Builder::should_visit_fields(AG::swift::metadata const*, bool) + 56
46 AttributeGraph                 0x57cc AG::LayoutDescriptor::Builder::visit_element(AG::swift::metadata const*, AG::swift::metadata::ref_kind, unsigned long, unsigned long) + 120
47 AttributeGraph                 0x5358 AG::swift::metadata_visitor::visit_field(AG::swift::metadata const*, AG::swift::field_record const&, unsigned long, unsigned long) + 180
48 AttributeGraph                 0x5220 AG::swift::metadata::visit(AG::swift::metadata_visitor&) const + 1244
49 AttributeGraph                 0x4c54 AG::LayoutDescriptor::make_layout(AG::swift::metadata const*, AGComparisonMode, AG::LayoutDescriptor::HeapMode) + 580
50 AttributeGraph                 0x4140 AG::(anonymous namespace)::TypeDescriptorCache::fetch(AG::swift::metadata const*, unsigned int, AG::LayoutDescriptor::HeapMode, int) + 344
51 AttributeGraph                 0x5928 AG::LayoutDescriptor::Builder::should_visit_fields(AG::swift::metadata const*, bool) + 56
52 AttributeGraph                 0x650c AG::LayoutDescriptor::Builder::visit_case(AG::swift::metadata const*, AG::swift::field_record const&, unsigned int) + 428
53 AttributeGraph                 0x511c AG::swift::metadata::visit(AG::swift::metadata_visitor&) const + 984
54 AttributeGraph                 0x4c54 AG::LayoutDescriptor::make_layout(AG::swift::metadata const*, AGComparisonMode, AG::LayoutDescriptor::HeapMode) + 580
55 AttributeGraph                 0x4140 AG::(anonymous namespace)::TypeDescriptorCache::fetch(AG::swift::metadata const*, unsigned int, AG::LayoutDescriptor::HeapMode, int) + 344
56 AttributeGraph                 0x5928 AG::LayoutDescriptor::Builder::should_visit_fields(AG::swift::metadata const*, bool) + 56
57 AttributeGraph                 0x57cc AG::LayoutDescriptor::Builder::visit_element(AG::swift::metadata const*, AG::swift::metadata::ref_kind, unsigned long, unsigned long) + 120
58 AttributeGraph                 0x5358 AG::swift::metadata_visitor::visit_field(AG::swift::metadata const*, AG::swift::field_record const&, unsigned long, unsigned long) + 180
59 AttributeGraph                 0x5220 AG::swift::metadata::visit(AG::swift::metadata_visitor&) const + 1244
60 AttributeGraph                 0x4c54 AG::LayoutDescriptor::make_layout(AG::swift::metadata const*, AGComparisonMode, AG::LayoutDescriptor::HeapMode) + 580
61 AttributeGraph                 0x4140 AG::(anonymous namespace)::TypeDescriptorCache::fetch(AG::swift::metadata const*, unsigned int, AG::LayoutDescriptor::HeapMode, int) + 344
62 AttributeGraph                 0x5928 AG::LayoutDescriptor::Builder::should_visit_fields(AG::swift::metadata const*, bool) + 56
63 AttributeGraph                 0x650c AG::LayoutDescriptor::Builder::visit_case(AG::swift::metadata const*, AG::swift::field_record const&, unsigned int) + 428
64 AttributeGraph                 0x511c AG::swift::metadata::visit(AG::swift::metadata_visitor&) const + 984
65 AttributeGraph                 0x4c54 AG::LayoutDescriptor::make_layout(AG::swift::metadata const*, AGComparisonMode, AG::LayoutDescriptor::HeapMode) + 580
66 AttributeGraph                 0x4140 AG::(anonymous namespace)::TypeDescriptorCache::fetch(AG::swift::metadata const*, unsigned int, AG::LayoutDescriptor::HeapMode, int) + 344
67 AttributeGraph                 0x5928 AG::LayoutDescriptor::Builder::should_visit_fields(AG::swift::metadata const*, bool) + 56
68 AttributeGraph                 0x57cc AG::LayoutDescriptor::Builder::visit_element(AG::swift::metadata const*, AG::swift::metadata::ref_kind, unsigned long, unsigned long) + 120
69 AttributeGraph                 0x5358 AG::swift::metadata_visitor::visit_field(AG::swift::metadata const*, AG::swift::field_record const&, unsigned long, unsigned long) + 180
70 AttributeGraph                 0x5220 AG::swift::metadata::visit(AG::swift::metadata_visitor&) const + 1244
71 AttributeGraph                 0x4c54 AG::LayoutDescriptor::make_layout(AG::swift::metadata const*, AGComparisonMode, AG::LayoutDescriptor::HeapMode) + 580
72 AttributeGraph                 0x4140 AG::(anonymous namespace)::TypeDescriptorCache::fetch(AG::swift::metadata const*, unsigned int, AG::LayoutDescriptor::HeapMode, int) + 344
73 AttributeGraph                 0x5928 AG::LayoutDescriptor::Builder::should_visit_fields(AG::swift::metadata const*, bool) + 56
74 AttributeGraph                 0x650c AG::LayoutDescriptor::Builder::visit_case(AG::swift::metadata const*, AG::swift::field_record const&, unsigned int) + 428
75 AttributeGraph                 0x511c AG::swift::metadata::visit(AG::swift::metadata_visitor&) const + 984
76 AttributeGraph                 0x4c54 AG::LayoutDescriptor::make_layout(AG::swift::metadata const*, AGComparisonMode, AG::LayoutDescriptor::HeapMode) + 580
77 AttributeGraph                 0x4140 AG::(anonymous namespace)::TypeDescriptorCache::fetch(AG::swift::metadata const*, unsigned int, AG::LayoutDescriptor::HeapMode, int) + 344
78 AttributeGraph                 0x5928 AG::LayoutDescriptor::Builder::should_visit_fields(AG::swift::metadata const*, bool) + 56
79 AttributeGraph                 0x57cc AG::LayoutDescriptor::Builder::visit_element(AG::swift::metadata const*, AG::swift::metadata::ref_kind, unsigned long, unsigned long) + 120
80 AttributeGraph                 0x5358 AG::swift::metadata_visitor::visit_field(AG::swift::metadata const*, AG::swift::field_record const&, unsigned long, unsigned long) + 180
81 AttributeGraph                 0x5220 AG::swift::metadata::visit(AG::swift::metadata_visitor&) const + 1244
82 AttributeGraph                 0x4c54 AG::LayoutDescriptor::make_layout(AG::swift::metadata const*, AGComparisonMode, AG::LayoutDescriptor::HeapMode) + 580
83 AttributeGraph                 0x1175c AG::(anonymous namespace)::TypeDescriptorCache::drain_queue(void*) + 348
84 libdispatch.dylib              0x3dd4 _dispatch_client_callout + 20
85 libdispatch.dylib              0x15a6c _dispatch_root_queue_drain + 864
86 libdispatch.dylib              0x1609c _dispatch_worker_thread2 + 156
87 libsystem_pthread.dylib        0x48f8 _pthread_wqthread + 228
88 libsystem_pthread.dylib        0x10cc start_wqthread + 8

com.google.firebase.crashlytics.MachExceptionServer
0  libsystem_kernel.dylib         0x1808 mach_msg2_trap + 8
1  libsystem_kernel.dylib         0x5008 mach_msg2_internal + 80
2  libsystem_kernel.dylib         0x4f20 mach_msg_overwrite + 436
3  libsystem_kernel.dylib         0x4d60 mach_msg + 24
4  REDACTED                       0x7fb18 FIRCLSMachExceptionServer + 194 (FIRCLSMachException.c:194)
5  libsystem_pthread.dylib        0x606c _pthread_start + 136
6  libsystem_pthread.dylib        0x10d8 thread_start + 8

com.apple.CFSocket.private
0  libsystem_kernel.dylib         0x9474 select$DARWIN_EXTSN + 8
1  libsystem_kernel.dylib         0x9474 __select + 8
2  CoreFoundation                 0xbeb7c __CFSocketManager + 640
3  libsystem_pthread.dylib        0x606c _pthread_start + 136
4  libsystem_pthread.dylib        0x10d8 thread_start + 8

com.apple.NSURLConnectionLoader
0  libsystem_kernel.dylib         0x1808 mach_msg2_trap + 8
1  libsystem_kernel.dylib         0x5008 mach_msg2_internal + 80
2  libsystem_kernel.dylib         0x4f20 mach_msg_overwrite + 436
3  libsystem_kernel.dylib         0x4d60 mach_msg + 24
4  CoreFoundation                 0x53f5c __CFRunLoopServiceMachPort + 160
5  CoreFoundation                 0x53600 __CFRunLoopRun + 1208
6  CoreFoundation                 0x52cd8 CFRunLoopRunSpecific + 608
7  CFNetwork                      0xfdc90 _CFHostIsDomainTopLevel + 108176
8  Foundation                     0xde718 __NSThread__start__ + 732
9  libsystem_pthread.dylib        0x606c _pthread_start + 136
10 libsystem_pthread.dylib        0x10d8 thread_start + 8

Thread
0  libsystem_pthread.dylib        0x10c4 start_wqthread + 8611840194

Thread
0  libsystem_pthread.dylib        0x10c4 start_wqthread + 8611840194

broadcast-ax-notify
0  libsystem_kernel.dylib         0x1808 mach_msg2_trap + 8
1  libsystem_kernel.dylib         0x5008 mach_msg2_internal + 80
2  libsystem_kernel.dylib         0x4f20 mach_msg_overwrite + 436
3  libsystem_kernel.dylib         0x4d60 mach_msg + 24
4  AXRuntime                      0x6570 _AXMIGBroadcastNotification + 464
5  AXRuntime                      0x6290 __AXPushNotificationToSystemForBroadcast_block_invoke_2 + 216
6  libdispatch.dylib              0x213c _dispatch_call_block_and_release + 32
7  libdispatch.dylib              0x3dd4 _dispatch_client_callout + 20
8  libdispatch.dylib              0xb400 _dispatch_lane_serial_drain + 748
9  libdispatch.dylib              0xbf30 _dispatch_lane_invoke + 380
10 libdispatch.dylib              0x16cb4 _dispatch_root_queue_drain_deferred_wlh + 288
11 libdispatch.dylib              0x16528 _dispatch_workloop_worker_thread + 404
12 libsystem_pthread.dylib        0x4934 _pthread_wqthread + 288
13 libsystem_pthread.dylib        0x10cc start_wqthread + 8

JavaScriptCore libpas scavenger
0  libsystem_kernel.dylib         0x71cc __psynch_cvwait + 8
1  libsystem_pthread.dylib        0x36e4 _pthread_cond_wait + 1228
2  JavaScriptCore                 0x155afe0 scavenger_thread_main + 1316
3  libsystem_pthread.dylib        0x606c _pthread_start + 136
4  libsystem_pthread.dylib        0x10d8 thread_start + 8

Thread
0  libsystem_pthread.dylib        0x10c4 start_wqthread + 8611840194

com.apple.root.user-initiated-qos
0  libsystem_kernel.dylib         0x1784 semaphore_wait_trap + 8
1  libdispatch.dylib              0x4370 _dispatch_sema4_wait + 28
2  libdispatch.dylib              0x4a20 _dispatch_semaphore_wait_slow + 132
3  REDACTED                       0x3351cc closure #1 in closure #1 in SessionImpl.exp() + 112 (SessionImpl.swift:112)
4  REDACTED                       0x3337e8 thunk for @escaping @callee_guaranteed @Sendable () -> () + 4303927272 (<compiler-generated>:4303927272)
5  libdispatch.dylib              0x213c _dispatch_call_block_and_release + 32
6  libdispatch.dylib              0x3dd4 _dispatch_client_callout + 20
7  libdispatch.dylib              0x15a6c _dispatch_root_queue_drain + 864
8  libdispatch.dylib              0x1609c _dispatch_worker_thread2 + 156
9  libsystem_pthread.dylib        0x48f8 _pthread_wqthread + 228
10 libsystem_pthread.dylib        0x10cc start_wqthread + 8

Thread
0  libsystem_pthread.dylib        0x10c4 start_wqthread + 8611840194

Thread
0  libsystem_pthread.dylib        0x10c4 start_wqthread + 8611840194

Thread
0  libsystem_pthread.dylib        0x10c4 start_wqthread + 8611840194
```
</details>

**Comment by orendayan** [comment]
We added a fix in our next release - 3.3.1, Thanks

**Comment by orendayan** [comment]
@msrutek-paylocity 

We've released v3.3.1, please upgrade and let us know if it solved the issues 
please reopen if needed

**Comment by msrutek-paylocity** [comment]
@orendayan Thanks for the info! Our release cycle takes about ~1 month until the change propagates its way to production. We'll reopen if needed.
